core/tests/testing-utils.ts

```diff
@@ -45,17 +49,44 @@ function createTestConfig (from, cascadingChanges): any {
   )
 }
 
-function createTestCloudFormationTemplate (stackDefinition = defaultCfTemplate): Template {
+export function createTestCloudFormationTemplate (stackDefinition = defaultCfTemplate): Template {
   return _.cloneDeep(stackDefinition)
 }
 
+export interface TemplateDashboardFinding {
+  dashboard: Dashboard
+  dashProperties: DashboardProperties
+}
+
+export function getDashboardFromTemplate (template: Template): TemplateDashboardFinding {
+  const dashResources = getResourcesByType('AWS::CloudWatch::Dashboard', template)
+  const [, dashResource] = Object.entries(dashResources)[0]
+  const dashboard = JSON.parse(dashResource.Properties?.DashboardBody['Fn::Sub'])
+  return {
+    dashboard,
+    dashProperties: dashResource.Properties as DashboardProperties
+  }
+}
+
+export function getDashboardWidgetsByTitle (dashboard: Dashboard, ...patterns: RegExp[]) {
+  return patterns.map((pattern) => {
+    const widgets = dashboard.widgets.filter((widget) => {
+      const props = widget.properties as MetricWidgetProperties
+      if (typeof props?.title === 'string') {
```

Comment:
maybe this ? to avoid the extra code flow 

```typescript
const widgets = dashboard.widgets.filter((widget) => {
      const props = widget.properties as MetricWidgetProperties
      return typeof props?.title === 'string' && pattern.test(props.title);
    })
```